worker/caasmodeloperator/modeloperator.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caasmodeloperator
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/names/v4"
+	"github.com/juju/utils"
+	"github.com/juju/version"
+	"gopkg.in/juju/worker.v1/catacomb"
+
+	"github.com/juju/juju/agent"
+	"github.com/juju/juju/api/caasmodeloperator"
+	"github.com/juju/juju/caas"
+)
+
+// ModelOperatorBroker describes the caas broker interface needed for installing
+// a ModelOperator into Kubernetes
+type ModelOperatorBroker interface {
+	EnsureModelOperator(string, string, *caas.ModelOperatorConfig) error
+}
+
+// ModelOperatorManager defines the worker used for managing model operators in
+// caas
+type ModelOperatorManager struct {
+	agentConfig agent.Config
+	api         caasmodeloperator.Interface
+	broker      ModelOperatorBroker
+	catacomb    catacomb.Catacomb
+	logger      Logger
+	modelUUID   string
+}
+
+const (
+	// DefaultModelOperatorPort is the default port used for the api server on
+	// the model operator
+	DefaultModelOperatorPort = 3300
```

Comment:
perhaps we could use something closer to the juju api port (17070), eg 17071 or 17080 or whatever